src/components/input/keyboardnavigation.js

```diff
@@ -146,6 +146,17 @@ define(["inputManager", "layoutManager"], function (inputManager, layoutManager)
         });
     }
 
+    // Gamepad initialisation. No script is required if no gamepads are present at init time, saving a bit of resources.
+    // Whenever the gamepad is connected, we hand all the control of the gamepad to gamepadtokey.js by removing the event handler
+    function attachGamepadScript(e) {
+        console.log("Gamepad connected! Attaching gamepadtokey.js script");
+        window.removeEventListener("gamepadconnected", attachGamepadScript);
+        require(["components/input/gamepadtokey"]);
```

Comment:
oh, wait, would this start input loop? maybe it would only load the script but not start the loop?

Reply:
That's why I need @dmitrylyzo and @MrTimscampi to test it.

Reply:
@JustAMan you are right.
* when I connect gamepad - no action
* press button "A" - `gamepadtokey.js` loaded, loop does not start
* any input - loop does not start

We have 2 options (both tested):
1. Load `gamepadtokey.js` in `site.js` - as in my variant somewhere above. Module will sleep until gamepad is connected.
2. Start loop automatically (at the very end of `gamepadtokey.js`) - as it was, but keep `startInputLoop` handler in case of reconnect gamepad.

Reply:
Tried this (not perfect but works):
```js
    function attachGamepad(e) {
        if (isGamepadConnected() && document.hasFocus()) {     <-------
            console.log("Gamepad connected! Starting input loop");
            startInputLoop();
        }
    }

    function dettachGamepad(e) {
        if (!isGamepadConnected() || !document.hasFocus()) {     <-------
            console.log("Gamepad disconnected! No other gamepads are connected, stopping input loop");
            stopInputLoop();
        } else {
            console.log("Gamepad disconnected! There are gamepads still connected.");
        }
    }

    // Event Listeners for any change in gamepads' state.
    window.addEventListener("gamepaddisconnected", dettachGamepad);
    window.addEventListener("gamepadconnected", attachGamepad);
    window.addEventListener("blur", dettachGamepad);     <-------
    window.addEventListener("focus", attachGamepad);    <-------
```
I think there was some issue with response to gamepad input, even if window is not focused.

Reply:
Cannot we call `attachGamepad()` from `gamepadtokey.js` when requiring the script upon first gamepad attach instead of options 1 and 2 above?

Reply:
> Cannot we call `attachGamepad()` from `gamepadtokey.js` when requiring the script upon first gamepad attach instead of options 1 and 2 above?

Yes, we can export it, but "2" is simpler. If we add what I posted above (focus-related), then we must run `attachGamepad` instead of `startInputLoop`.